src/main/java/com/bmc/truesight/saas/remedy/integration/impl/CallableBulkEventHttpClient.java

```diff
@@ -0,0 +1,173 @@
+import com.bmc.truesight.saas.remedy.integration.util.Constants;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class CallableBulkEventHttpClient implements Callable<Result>, BulkEventHttpClient {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CallableBulkEventHttpClient.class);
+
+    public CallableBulkEventHttpClient(List<TSIEvent> eventList, Configuration configuration) {
+        this.eventList = eventList;
+        this.configuration = configuration;
+    }
+
+    private List<TSIEvent> eventList;
+    private Configuration configuration;
+
+    @Override
+    public Result call() throws Exception {
+        return pushBulkEventsToTSI(eventList);
+    }
+
+    private String convertStreamToString(InputStream instream) {
+        BufferedReader reader = new BufferedReader(new InputStreamReader(instream));
+        StringBuilder sb = new StringBuilder();
+        String line = null;
+        try {
+            while ((line = reader.readLine()) != null) {
+                sb.append(line + "\n");
+            }
+        } catch (IOException e) {
+            LOG.debug("Exception in converting input stream to String, {}", e.getMessage());
+        } finally {
+            try {
+                instream.close();
+            } catch (IOException e) {
+                LOG.debug("Exception in closing the input stream, {}", e.getMessage());
+            }
+        }
+        return sb.toString();
+    }
+
+    public static String encodeBase64(final String encodeToken) {
+        byte[] encoded = Base64.encodeBase64(encodeToken.getBytes());
+        return new String(encoded);
+    }
+
+    @Override
+    public Result pushBulkEventsToTSI(List<TSIEvent> bulkEvents) throws BulkEventsIngestionFailedException {
+        LOG.debug("Starting ingestion of {} events  to TSI ", bulkEvents.size());
+        if (bulkEvents.size() <= 0) {
+            throw new BulkEventsIngestionFailedException("Cannot send empty events list to TSI");
+        }
+        Result result = null;
+        HttpClient httpClient = null;
+        boolean isSuccessful = false;
+        int retryCount = 0;
+        while (!isSuccessful && retryCount <= this.configuration.getRetryConfig()) {
+            httpClient = HttpClientBuilder.create().build();
+            HttpPost httpPost = new HttpPost(this.configuration.getTsiEventEndpoint());
+            httpPost.addHeader("Authorization", "Basic " + encodeBase64("" + ":" + this.configuration.getTsiApiToken()));
+            httpPost.addHeader("Content-Type", "application/json");
+            httpPost.addHeader("accept", "application/json");
+            httpPost.addHeader("User-Agent", "RemedyScript");
+            ObjectMapper mapper = new ObjectMapper();
+            String jsonInString = null;
```

Comment:
I guess, you can initialize httpPost outside loop and reuse the same object every time with new entity

what say?